Convert editor demos to ts (#2738)
DevExpress/devextreme-demos: README.md

1. Use the built-in CLI to choose or enter the category, the demo name, and the technology for the new demo.

> ### TS React infrastructure

Reviewer: ```suggestion
### TS React Infrastructure
```

> 1. After you make any changes in react typescript sources you should run the following command:  

Reviewer: ```suggestion
1. After you make any changes in React TypeScript sources, run the following command:  
```

```
npm run convert-to-js ./JSDemos/Demos/**/React
```

> 1. To ensure that React JavaScript sources are always in sync, there is a workflow: "Check generated JS demos"

Reviewer: ```suggestion
1. To ensure that React JavaScript and TypeScript sources are always in sync, the following GitHub action is used: "Check generated JS demos".
```

> 1. To ensure, that react javascript sources are always in sync, there is a workflow: "Check generated JS demos"

Reviewer: ```suggestion
1. To ensure that React JavaScript sources are always in sync, there is a workflow: "Check generated JS demos"
```

In sync with what? And is this workflow on GitHub or where?

Reviewer 2: in sync with typescript sources)
now the sources are in typescript and js files are just transpiled ts
this workflow is on github actions lint -> check_generated_demos
